frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,49 @@
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
-
+export const ConfigMapData = ({data}) => {
   const dl = [];
-  Object.keys(data || []).sort().forEach(k => {
+  Object.keys(data || {}).sort().forEach(k => {
     dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
+    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
   });
 
   return <dl>{dl}</dl>;
 };
 
-export default ConfigMapAndSecretData;
+export class SecretData extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { showSecret: false };
+    this.toggleSecret = this.toggleSecret.bind(this);
+  }
+
+  toggleSecret() {
+    this.setState({ showSecret: !this.state.showSecret });
+  }
+
+  render() {
+    const { data } = this.props;
+    const { showSecret } = this.state;
+    const dl = [];
+    Object.keys(data).sort().forEach(k => {
+      dl.push(<dt key={`${k}-k`}>{k}</dt>);
+      dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{showSecret ? window.atob(data[k]) : '*****'}</pre></dd>);
```

Comment:
Wondering if a dot &bull; would look nicer than `*`. Something like (untested)

```jsx
// Mask values with 5 dots.
const masked = <React.Fragment>
  <span className="sr-only">value hidden</span>
  <span aria-hidden="true">&bull;&bull;&bull;&bull;&bull;</span>
</React.Fragement>;

// ...

{showSecret ? window.atob(data[k]) : masked}
```

Probably need to remove the newlines if it's in a pre block though